### PR TITLE
Adding Support for oneColumnModeSort Options (Custom Responsive Sort Order)

### PR DESCRIPTION
**Pull request by gooeyideas**
…
### Description
These are changes implementing the requested changes from issue #713.

oneColumnModeSort option for sorting that also respects the existing `rtl` property.

`oneColumnModeSort: "row | row-reverse | column | column-reverse | none"`

- `none`: Sort by the order the widget's appear in the markup
- `row`:  (default) Sort first by **y** values (low to high) then sort widgets with the same y value by their **x** value (based on the `rtl` setting)
- `row-reverse`: Sort first by **y** values (high to low) then sort widgets with the same **y** value by their **x** value (based on the `rtl` setting)
- `col`: Sort first by **x** values (based on the `rtl` setting) then sort widgets with the same **x** value by their **y** value (low to high)
- `col-reverse`: Sort first by **x** values (based on the `rtl` setting) then sort widgets with the same **x** value by their **y** value (high to low)

Please explain the changes you made here. Include an example of what your changes fix or how to use the changes: **Please refer to #713 for more info**

### Checklist
- [X] Created tests which fail without the change (if possible)
- [X] All tests passing (`npm test`)
- [ ] Extended the README / documentation, if necessary
**I am not sure how to update the doc/README.md with the auto jsdoc generation tool. Is there anything else I need to add?**

### CODE REVIEW

File: src/gridstack.js [hunk 1]
```diff
@@ -677,9 +681,15 @@
             cellHeightUnit: 'px',
             disableOneColumnMode: opts.disableOneColumnMode || false,
             oneColumnModeClass: opts.oneColumnModeClass || 'grid-stack-one-column-mode',
+            oneColumnModeSort: opts.oneColumnModeSort || 'row',
             ddPlugin: null
         });
 
+        this.opts.oneColumnModeSort = this.opts.oneColumnModeSort.toLowerCase();
+        if (!String('row | row-reverse | column | column-reverse | none').includes(this.opts.oneColumnModeSort)) {
+            this.opts.oneColumnModeSort = 'row';
```
Comment (adumesny) [inline src/gridstack.js]: nice to keep original sort default. It's the most logical one really.

File: src/gridstack.js [hunk 2]
```diff
@@ -777,19 +787,25 @@
                 }
                 self.container.addClass(self.opts.oneColumnModeClass);
                 oneColumnMode = true;
+                var oneColumnModeSort = self.opts.oneColumnModeSort;
+                if (oneColumnModeSort !== 'none') {
+                    var dim = oneColumnModeSort.includes('column') ? -1 : 1;
+                    var dir = dim > 0 && self.opts.rtl ? -1 : 1;
+                    dir = oneColumnModeSort.includes('reverse') ? dir * -1 : dir;
```
Comment (adumesny) [inline src/gridstack.js]: code could use some comment explanation. rather hard to parse. for one thing you compare dim > 0, yet set value 1:-1 above so why not use those 2 values ?

**Comment by adumesny** [comment]
"I am not sure how to update the doc/README.md with the auto jsdoc generation tool. Is there anything else I need to add?"
I believe README.md is manually updated - that's what I've been doing. Format is pretty self explanatory (and some tools like Visual Studio Code will render a pretty form of it so you can verify your tag formatting before checking it in).

**Comment by gooeyideas** [comment]
Updated the original PR but I still need to go in and update the README, I will try to find some time to do that next week.

**Comment by adumesny** [comment]
@gooeyideas any chance you might update this PR to the current code base ? we're actively accepting review and releasing frequent builds right now...

**Comment by adumesny** [comment]
@gooeyideas I completely changed how we position widgets when changing column sizing (1 being the easy case really) with #1098. While this doesn't invalidate the need to have 'custom' order (dom vs row which is what I do hopefully corecly now) you may wan to check the latest behavior. And adjust your review if you still feel the need to have even more control...
having saved layouts could also mix with that you have (way for user to set those as opposed to gridstack generating them)

**Comment by adumesny** [comment]
@gooeyideas please update the review. I don't think we should support all the modes, but rather having 1 column use left-to-right-top-bottom (default) or DOM order (so you can specify a very different 1 column layout (ideally with hidden items as well).

The other directions (RTL for example) should be tied to a generic gravity that affects all things (drag&drop, moving) rather than one off 1column thing IMO.

I would love to give you credit for the DOM order at least.

**Comment by gooeyideas** [comment]
Sure thing, for missing the last few messages. I recently discovered that my company changed some of our email rules so that github alerts were being treated as spam. I will try get around to this before the end of this week. Sorry again.

**Comment by adumesny** [comment]
great I look forward to it! are you ok doing just the DOM vs default row order we have now and we can add the others later or tie them to a generic gravity of the grid settings ?

I'm thinking that DOM order makes sense as way to specify a custom 1 column layout, whereas x,y are used for multi column layouts.

I've changed the code to support a robust multi column layout where we store each config (so going from 12 to 1 (or anything in between) and back would preserve the original layout, but have not worked on oneColumn yet, and was hoping to get your change in before I do.

**Comment by adumesny** [comment]
@gooeyideas supporting just DOM order (vs default row sorting) is getting much trickier than I expected now that `oneColumnMode` is just `setColumn(1)` to allow full editing (resize, move, insert) and better heuristics as we don't re-order dom elements during editing so changes will not last.

I may try in sandbox but punt on this until we support per column layout instead in the file format (what I create in memory).

**Comment by adumesny** [comment]
I didn't hear back so closing this as I ended up doing #1126 instead. Thanks for the initial work and let me know (in #713) if I missed anything
